compiler-core/src/type_/error.rs

```diff
@@ -333,6 +333,11 @@ pub enum Warning {
     UnnecessaryDoubleBoolNegation {
         location: SrcSpan,
     },
+
+    PerfListLength {
```

Comment:
This is just a placeholder name for the warning while I think of something better.

Suggestions welcome!